<launch>
    
    <!-- Spawn Task-2 Models in Gazebo -->
    <include file="$(find pkg_vb_sim)/launch/task4_world.launch" />

    <!-- Spawn two UR5 in Gazebo -->
    <include file = "$(find pkg_vb_sim)/launch/two_ur5_gazebo.launch" />

    <!-- Launch two move_group nodes for the two UR5 Arms -->
    <include file="$(find pkg_vb_sim)/launch/two_ur5_move_group.launch" />
    
    <!-- Spawn T4 Boxes -->
    <node name= "task4_spawn_models" pkg= "pkg_vb_sim" type="task4_spawn_models.py"/>

    <!-- Run Conveyor Belt Service Server -->
    <node name= "node_service_server_conveyor_belt" pkg= "pkg_vb_sim" type="node_service_server_conveyor_belt.py" output="screen"/>
	
    <!-- Thia launches ur5_1 node-->
    <node name= "ur5_1" pkg= "pkg_task4" type="ur5_1.py" output="screen"/>
    <!-- Thia launches ur5_2 node-->
    <node name= "ur5_2" pkg= "pkg_task4" type="ur5_2.py" output="screen"/>

    <!-- Recording Bag File for Submission -->
    <arg name="record" default="false"/>
    <arg name="rec_name" default="task4.bag"/>

    <group if="$(arg record)">
        <node name="rosbag_record_pick1" pkg="rosbag" type="record" args="record -O $(find pkg_task4)/bag_files/$(arg rec_name) --chunksize=10 /eyrc/vb/ur5_1/vacuum_gripper/logical_camera/ur5_1 /eyrc/vb/ur5_2/vacuum_gripper/logical_camera/ur5_2" output="screen"/>
    </group>


</launch>